modules/new_profile/js/tmp_hash.js

```diff
@@ -0,0 +1,85 @@
+$(document).ready(function () {
+  $("#new_profile").submit(function (event) {
+    event.preventDefault();
+    $("[name=fire_away]").prop('disabled',true);
+    if (!confirm('Personal data cannot be changed once entered. Are you sure all data is correct?')) {
+      return;
+    }
+    //generate the GUID
+    var first_name = $("[name=first_name]").val(),
+      middle_name = $("[name=middle_name]").val(),
+      last_name = $("[name=last_name]").val(),
+      city_of_birth = $("[name=city_of_birth]").val(),
+      date_of_birth = $("[name=dob1]").val(),
+      gender = $("[name=gender]").val();  //todo: ensure acceptable for hash algo, loris, pii
+
+    var OpenScienceIdentity = require('open-science-identity').OpenScienceIdentity;
+    var id = new OpenScienceIdentity({
+      first_name: first_name,
+      middle_name: middle_name,
+      last_name: last_name,
+      city_of_birth: city_of_birth,
+      birth_day: date_of_birth,
+      gender: gender
+    });
+    try {
+      var guid = id.toSignature();
```

Comment:
Unless I'm missing something this is just a hash of personal information, not a guid.

Reply:
yes, we've just been calling it GUID until now

Reply:
Yeah GUID has been thrown around as an unfortunate misnomer. The idea is to create a unique hash based on a minimal set of PII. It has nothing to do with being a GUID in the sense that it's used in databasing.

Reply:
I can call it UID if it makes you guys feel better.